runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/io/DataflowDatastoreIOTest.java

```diff
@@ -1,66 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.beam.runners.dataflow.io;
-
-import static org.apache.beam.sdk.transforms.display.DisplayDataMatchers.hasDisplayItem;
-
-import static org.hamcrest.Matchers.hasItem;
-import static org.junit.Assert.assertThat;
-
-import org.apache.beam.runners.dataflow.transforms.DataflowDisplayDataEvaluator;
-import org.apache.beam.sdk.io.datastore.DatastoreIO;
-import org.apache.beam.sdk.transforms.PTransform;
-import org.apache.beam.sdk.transforms.display.DisplayData;
-import org.apache.beam.sdk.transforms.display.DisplayDataEvaluator;
-import org.apache.beam.sdk.values.PBegin;
-import org.apache.beam.sdk.values.PCollection;
-import org.apache.beam.sdk.values.POutput;
-
-import com.google.datastore.v1beta3.Entity;
-import com.google.datastore.v1beta3.Query;
-
-import org.junit.Test;
-
-import java.util.Set;
-
-/**
- * Unit tests for Dataflow usage of {@link DatastoreIO} transforms.
- */
-public class DataflowDatastoreIOTest {
-  @Test
-  public void testSourcePrimitiveDisplayData() {
```

Comment:
In reference to [my pending change to move these to `@RunnableOnService`](https://github.com/apache/incubator-beam/pull/607) and the test failure above: in this case the display data integration is broken in the default runner-agnostic implementation, only Dataflow runner retains the primitive display data.

Specifically, `Combine.GroupedValues` [implements `populateDisplayData()`] (https://github.com/apache/incubator-beam/blob/1963bde446a798f0971393bb8f0076fc933367cf/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java#L2359) as expected, but [the `PTransform.apply` method](https://github.com/apache/incubator-beam/blob/1963bde446a798f0971393bb8f0076fc933367cf/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java#L2285) is implemented in terms of lower-level transforms, and these leaf nodes do not retain the display data. In the Dataflow runner, [`Combine.GroupedValues` becomes a primitive](https://github.com/apache/incubator-beam/blob/1963bde446a798f0971393bb8f0076fc933367cf/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java#L811), so its display data is retained through graph translation.

So, my 2 cents:

1. These tests are valuable because they verify runners retain display data through translation, which is otherwise difficult to verify manually.
2. The default display data implementation for `Combine.GroupedValues` is broken and should be fixed.
3. The whole process for passing composite display data down to leaf nodes is terrible, and I can't wait until we have composite PTransforms represented in the runner API.